sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java

```diff
@@ -217,26 +218,39 @@ protected AvroCoder(Class<T> type, Schema schema) {
     this.decoder = new EmptyOnDeserializationThreadLocal<>();
     this.encoder = new EmptyOnDeserializationThreadLocal<>();
 
-    // Reader and writer are allocated once per thread and are "final" for thread-local Coder
-    // instance.
-    this.reader = new EmptyOnDeserializationThreadLocal<DatumReader<T>>() {
+    // ReflectData, Reader and writer are allocated once per thread per Coder
+    this.reflectData = new EmptyOnDeserializationThreadLocal<ReflectData>() {
```

Comment:
Done.

I didn't necessarily trust `ReflectData` since it is basically undocumented, but since it was a singleton I guess it is implied.